devices/r4isdhc.cpp

```diff
@@ -326,6 +339,8 @@ class R4iSDHC : Flashcart {
             return false;
         }
 
+        // type2 carts read 0x8000-0x10000 from 0x1F8000-0x200000 instead of from 0x8000
+        uint32_t firm_header_size = cart_type == 1 ? 0x200 : 0x8200;
```

Comment:
`firm_header_size` doesn't make sense, because if `cart_type == 2` then this isn't just the FIRM header anymore.

I'd rather you just inline this.